### PR TITLE
add e2e for vault provider

### PR DESCRIPTION
**Pull request by sozercan**

### CODE REVIEW

File: pkg/providers/vault/examples/nginx-pod-vault-inline-volume.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ spec:
         volumeAttributes:
           providerName: "vault"
           roleName: "example-role"
-          vaultAddress: "http://10.0.146.70:8200"
+          vaultAddress: "http://10.97.45.20:8200"
```
Comment (ritazh) [inline pkg/providers/vault/examples/nginx-pod-vault-inline-volume.yaml]: Does this need to be updated?

File: test/bats/tests/nginx-pod-vault-inline-volume-secretproviderclass.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    - name: secrets-store-inline
+      mountPath: "/mnt/secrets-store"
+      readOnly: true
+  volumes:
+    - name: secrets-store-inline
+      csi:
+        driver: secrets-store.csi.k8s.com
+        readOnly: true
+        volumeAttributes:
+          secretProviderClass: "vault-foo"
```
Comment (ritazh) [inline test/bats/tests/nginx-pod-vault-inline-volume-secretproviderclass.yaml]: nit: newline

File: test/bats/tests/vault_v1alpha1_secretproviderclass.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  parameters:
+    roleName: "example-role"
+    vaultAddress: http://${VAULT_SERVICE_IP}:8200
+    vaultSkipTLSVerify: "true"
+    objects:  |
+      array:
+        - |
+          objectPath: "/foo"
+          objectName: "bar"
+          objectVersion: ""
```
Comment (ritazh) [inline test/bats/tests/vault_v1alpha1_secretproviderclass.yaml]: nit: newline

File: test/bats/vault.bats [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    capabilities = ["read", "list"]
+  }
+
+  path "sys/renew/*" {
+    capabilities = ["update"]
+  }
+EOF
+  assert_success
+
+  run kubectl exec -it $VAULT_POD -- vault write auth/kubernetes/role/example-role \
```
Comment (ritazh) [inline test/bats/vault.bats]: Nice!

File: pkg/providers/vault/examples/nginx-pod-vault-inline-volume.yaml (hunk 1 above)
Comment (sozercan) [reply pkg/providers/vault/examples/nginx-pod-vault-inline-volume.yaml]: this is an accidental commit when i was testing, will revert

**Pull request merged**